src/meta/meta_backup_service.h

```diff
@@ -97,58 +92,23 @@ struct backup_start_time
            << std::setfill('0') << std::to_string(minute);
         return ss.str();
     }
-    // NOTICE: this function will modify hour and minute, if time is invalid, this func will set
-    // hour = 24, minute = 0
+
+    // If start_time is valid, return true.
     bool parse_from(const std::string &time)
     {
-        if (::sscanf(time.c_str(), "%d:%d", &hour, &minute) != 2) {
+        if (sscanf(time.c_str(), "%d:%d", &hour, &minute) != 2) {
             return false;
-        } else {
-            if (hour > 24) {
-                hour = 24;
-                minute = 0;
-                return false;
-            }
-
-            if (hour == 24 && minute != 0) {
-                minute = 0;
-                return false;
-            }
-
-            if (minute >= 60) {
-                hour = 24;
-                minute = 0;
-                return false;
-            }
         }
-        return true;
-    }
-
-    // return the interval between new_hour:new_min and start_time,
-    // namely new_hour:new_min - start_time;
-    // unit is ms
-    int64_t compute_time_drift_ms(int32_t new_hour, int32_t new_min)
-    {
-        int64_t res = 0;
-        // unit is hour
-        res += (new_hour - hour);
-        // unit is minute
-        res *= 60;
-        res += (new_min - minute);
-        // unit is ms
-        return (res * 60 * 1000);
+        return (hour >= 0 && hour < 24 && minute >= 0 && minute < 60);
     }
 
-    // judge whether we should start backup base current time
     bool should_start_backup(int32_t cur_hour, int32_t cur_min)
     {
-        if (hour == 24) {
-            // erase the restrict of backup_start_time, just for testing
+        if (hour == 24 && minute == 0) {
+            // only for tests
             return true;
         }
-        // NOTICE : if you want more precisely, you can use cur_min to implement
-        // now, we just ignore
-        return (cur_hour == hour);
+        return (cur_hour == hour) && (cur_min == minute);
```

Comment:
This raises me a question, backup used to have a problem that if backup interval less than one day, it will backup without control, it is caused by the hour check in function `should_start_backup`. Is my understanding correct? Besides, if we check hour and minute here, is that mean backup interval should always be 1 day or  integer multiple of days?

Reply:
We only check this when we want to start the first time backup.